runtime/jcl/common/thread.cpp

```diff
@@ -571,10 +571,15 @@ Java_java_lang_VirtualThread_notifyJvmtiMountBegin(JNIEnv *env, jobject thread,
 			J9OBJECT_OBJECT_STORE(currentThread, rootPrev, vm->virtualThreadLinkNextOffset, threadObj);
 			J9OBJECT_OBJECT_STORE(currentThread, root, vm->virtualThreadLinkPreviousOffset, threadObj);
 		}
-		omrthread_monitor_exit(vm->liveVirtualThreadListMutex);
-
-		vmFuncs->internalExitVMToJNI(currentThread);
+	} else {
+		/* If this virtual thread is being inspected, do not allow a yielded thread to be resumed. */
+		while (0 != J9OBJECT_U64_LOAD(currentThread, threadObj, vm->virtualThreadInspectorCountOffset)) {
+			omrthread_monitor_wait(vm->liveVirtualThreadListMutex);
```

Comment:
you can `pushObjectInSpecialFrame` then `releaseVMAccess` before the wait and do the opposite after the wait

Reply:
that being said, Im not sure this is the correct approach. I think you can use haltThreadForInspection if the vthread is mounted, and introduce an equivalent for unmounted threads on j9vmcontinuations. THat would be a more general approach, and I think that would benefit us because we will need something like Thread::getStateImpl and Continuation::getStackTrace

Reply:
perhaps you can just pin thread, and add a check in continuationEnter, that if its pinned it can't enter

Reply:
- It is identical to the blocking for j.l.Thread:
https://github.com/eclipse-openj9/openj9/blob/984612f68ad11944e7ce1ecec7b57cb9a7f6eaee/runtime/vm/jvmfree.c#L200-L205

- A running virtual thread is not halted with these changes. It still runs. We only block it from unmounting similar to the above changes.
- We have to do the same for a yielded thread. We only block it from being mounted/running again until JVMTI finishes inspecting it in the yielded state. 
- Virtual thread invokes `Continuation::getStackTrace` in an unmounted state after setting the `SUSPENDED` flag : [VirtualThread.java#L862-L865](https://github.com/ibmruntimes/openj9-openjdk-jdk19/blob/703f7587b12c993e641847e9d0c5402497ef9afb/src/java.base/share/classes/java/lang/VirtualThread.java#L862-L865). The `SUSPENDED` flag prevents it to be rescheduled. So, no blocking or synchronization is needed for `Continuation::getStackTrace`.

Reply:
> you can pushObjectInSpecialFrame then releaseVMAccess before the wait and do the opposite after the wait

Updated.

Reply:
Okay, perhaps this is the best way. I was hoping there was a way to reuse the pinning mechanism for this, but I don't think there is.